RecoParticleFlow/PFRecHitProducer/plugins/alpaka/PFRecHitSoAProducer.cc

```diff
@@ -53,10 +53,13 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
       edm::ParameterSetDescription desc;
       edm::ParameterSetDescription producers;
-      producers.add<edm::InputTag>("src")->setComment("Input CaloRecHitSoA");
-      producers.add<edm::ESInputTag>("params")->setComment("Quality cut parameters");
-      desc.addVPSet("producers", producers)->setComment("List of inputs and quality cuts");
-      desc.add<edm::ESInputTag>("topology")->setComment("Topology information");
+      producers.add<edm::InputTag>("src", edm::InputTag(""))->setComment("Input CaloRecHitSoA");
+      producers.add<edm::ESInputTag>("params", edm::ESInputTag(""))->setComment("Quality cut parameters");
+      std::vector<edm::ParameterSet> producersDefault(1);
+      producersDefault[0].addParameter<edm::InputTag>("src", edm::InputTag(""));
+      producersDefault[0].addParameter<edm::ESInputTag>("params", edm::ESInputTag(""));
```

Comment:
Example of one of the auto-generated `cfi` files for this plugin in `CMSSW_14_0_0_pre3`.
```python
import FWCore.ParameterSet.Config as cms

pfRecHitSoAProducerHCAL = cms.EDProducer('PFRecHitSoAProducerHCAL@alpaka',
  producers = cms.VPSet(
    cms.PSet(
      src = cms.InputTag('')
    )
  ),
  topology = cms.ESInputTag('', ''),
  synchronise = cms.untracked.bool(False),
  mightGet = cms.optional.untracked.vstring,
  alpaka = cms.untracked.PSet(
    backend = cms.untracked.string('')
  )
)
```
I could not get `params` to be added under `producers`, despite the fact that its default value is specified in `fillDescriptions`, so I am missing something.

It might be that this is not an issue for ConfDB parsing, if nested PSets like this one can be edited in ConfDB regardless of these defaults parameters. I'm not 100% sure this is the case, though.

If I change the type of `params` from `edm::ESInputTag` to something else (I tried `std::string`), then it does appear as intended inside the `producers` VPSet of the auto-generated cfi file.

Reply:
As mentioned in mattermost, this should be sufficient for the ConfDB side.